session_5/readme/README.md

```diff
@@ -0,0 +1,159 @@
+SCHEMA
+```ruby
+ActiveRecord::Schema.define(version: 2021_01_30_150003) do
+
+  create_table "assignments", force: :cascade do |t|
+    t.string "name"
+    t.decimal "weightage"
+    t.datetime "deadline"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.integer "course_id"
+    t.index ["course_id"], name: "index_assignments_on_course_id"
+  end
+
+  create_table "branches", force: :cascade do |t|
+    t.string "name"
+    t.string "code"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.integer "course_id"
+    t.integer "student_id"
+    t.index ["course_id"], name: "index_branches_on_course_id"
+    t.index ["student_id"], name: "index_branches_on_student_id"
+  end
+
+  create_table "courses", force: :cascade do |t|
+    t.string "name"
+    t.string "code"
+    t.integer "year"
+    t.integer "credits"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.integer "branch_id"
+    t.integer "student_id"
+    t.integer "assignment_id"
+    t.index ["assignment_id"], name: "index_courses_on_assignment_id"
+    t.index ["branch_id"], name: "index_courses_on_branch_id"
+    t.index ["student_id"], name: "index_courses_on_student_id"
+  end
+
+  create_table "students", force: :cascade do |t|
+    t.string "name"
+    t.integer "admission_year"
+    t.string "email"
+    t.string "roll_no"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.integer "branch_id"
+    t.integer "course_id"
+    t.index ["branch_id"], name: "index_students_on_branch_id"
+    t.index ["course_id"], name: "index_students_on_course_id"
+  end
+
+end
+```
+VALIDATIONS & ASSOCIATIONS
+```ruby
+class Branch < ApplicationRecord
+    has_many :courses
+    has_many :students
+
+    validates :name, presence: true, inclusion: {
+        :in => ['Computer Science and Engineering', 'Mechanical Engineering', 'Mining Engineering', 'Electronics and Communication Engineering'],
+        message: "%{value} is not a valid branch."
+    }
+    
+    validates :code, presence: true, uniqueness: true
+end
+```
+```ruby
+class Course < ApplicationRecord
+    belongs_to :branch
+    has_many :students
+    has_many :assignments
+
+    validates_associated :branch
+
+    validates :name, presence: true
+
+    validates :code, presence: true, uniqueness: true
+    validate :check_code
```

Comment:
The custom validations functions are named `{attribute_name}_cannot_{condition}` or `{attribute_name}_must_{condition}`, so you can split this into more functions and call them - `code_must_start_with_branch_code` and `code_must_have_three_digits`.

When I read function name `check_code`, I have no idea what the function does but `code_must_start_with_branch_code` is pretty self-explanatory.